pkg/apis/v1beta1/ec2nodeclass.go

```diff
@@ -152,6 +176,30 @@ type AMISelectorTerm struct {
 	Owner string `json:"owner,omitempty"`
 }
 
+// LicenseSelectorTerm defines selection logic for a licenseConfigurationSpecification used to launch nodes
+// If multiple fields are used for selection, the requirements are ANDed.
+type LicenseSelectorTerm struct {
```

Comment:
Should we consider doing ID or ARN for any of these? I'm curious generally if we are going to make all of these terms if there's a use-case for allowing the LaunchTemplate request from a set of options, rather than just specifying a single option. 

I'm not sure I have enough depth on the topic right now to make that call, though I'm happy to look into it a bit deeper if we need help making that call here

Reply:
I've stayed away from IDs and ARNs as I feel they're counter to the ethos of the selector pattern. If we want different clusters / regions / accounts to be able to share the same configuration, name is the most portable selector.

My currently thinking is we should implement the selectors with name only first, and then we can easily extend them to select based upon different identifiers as needed.